scipy/_build_utils/src/npy_cblas.h

```diff
@@ -15,6 +15,18 @@ extern "C"
 {
 #endif
 
+/*
+ * GCC and Clang require functions from static libraries to have this attribute
+ * in order for the linker to strip them if they are unused during linkage of
+ * dynamic libraries. We require GCC > 4 and Clang defines __GNUC__ as 4. MSVC
+ * does not require this macro because it makes all symbols hidden by default.
```

Comment:
A bit more details here, so that future readers don't have to go digging into the PR comments.
```suggestion
 * dynamic libraries, see https://gcc.gnu.org/wiki/Visibility.
 * This is supported since GCC 4.1 (though our lowest supported version is much
 * higher already), which we keep using here since Clang defines __GNUC__ as 4
 * (since at least Clang 2.7 and up to the most recent Clang 18), see
 * https://clang.llvm.org/docs/UsersManual.html#cmdoption-fgnuc-version
 * MSVC does not require this macro because it hides all symbols by default.
```

Reply:
Did away with this complexity by using the `gnu_symbol_visibility` Meson option.